main_ex3.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+func main() {
+	for {
+		choice := mainMenu()
+		if choice == 1 {
+			listEntries()
+		} else if choice == 2 {
+			add()
+		} else if choice == 4 {
+			break
+		} else {
+			fmt.Println("ERR: wrong choice", choice)
+		}
+	}
+}
+
+func mainMenu() (choice int) {
+	fmt.Println(`Choose your action:
+1) List entries
+2) Add new
+3) Remove by ID
+4) Quit`)
+	fmt.Scanln(&choice)
+	return choice
+}
+
+type Entry struct {
+	ID          uint32
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func printHeader() {
+	fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+	fmt.Println("-------------------------------------------------------------------------")
+}
+
+const (
+	phonebookFile = "phonebook.json"
+	pageLines     = 20
+)
+
+func listEntries() {
+	var entries []Entry
+
+	f, err := os.Open(phonebookFile)
```

Comment:
Please factoring out this code to a dedicated function. Just to ensure you don't have to repeat it. It might look smth like this here:
```go
entries, err := LoadPhonebook()
if err != nil {
  log.Fatal(err)
}
```
instead of this whole reading and decoding